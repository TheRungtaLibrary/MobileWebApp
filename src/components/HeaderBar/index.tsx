import React from 'react';
import Box from '@mui/material/Box';
import Toolbar from '@mui/material/Toolbar';
import {
    StyledAppBar,
    Logo
} from './styles';

const HeaderBar = (): JSX.Element => {
    return (
        <Box sx={{ flexGrow: 1 }}>
            <StyledAppBar position="static">
                <Toolbar variant="dense">
                    <Box sx={{ flexGrow: 16 }} >
                        <Logo>Car Photos</Logo>
                    </Box>
                </Toolbar>
            </StyledAppBar>
        </Box >
    );
}

export default HeaderBar;
