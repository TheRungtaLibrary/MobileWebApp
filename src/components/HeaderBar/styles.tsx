import styled from 'styled-components';
import AppBar from '@mui/material/AppBar';
import Typography from '@mui/material/Typography';

export const StyledAppBar = styled(AppBar)(({ theme }) =>
    `
    background: white;
    color: ${theme.palette.background.light};
    box-shadow: none;
    border-bottom: solid 1px ${theme.palette.border.main};
    height: 3rem;
    `
);

export const Logo = styled(Typography)`
    font-family: "DomaineDisplayNarrow", "Times New Roman", serif;
    margin-right: 1rem;
    white-space: nowrap;
    justify-content: center;
    font-size: 2rem;
`

