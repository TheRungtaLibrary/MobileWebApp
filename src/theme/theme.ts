export const theme = {
    palette: {
        primary: {
            main: '#0052cc',
        },
        secondary: {
            main: '#edf2ff',
        },
        tertiary: {
            main: '#F8F8F8',
            light: '#ffffff',
        },
        background: {
            main: '#f6ded2',
            light: '#000',
        },
        border: {
            main: '#efefef',
            light: '#808080',
        },
        text: {
            main: '#ebedf2',
        }
    },
};