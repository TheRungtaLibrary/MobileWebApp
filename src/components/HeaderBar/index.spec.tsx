import React, { FunctionComponent } from 'react';
import { render } from '@testing-library/react';
import HeaderBar from './index';
import { ThemeProvider } from 'styled-components';
import { theme } from '../../theme/theme';

describe('Given <HeaderBar />', () => {

    let component: HTMLElement;
    let Page: FunctionComponent<any>;

    beforeEach(() => {
        Page = (): JSX.Element => (
            <ThemeProvider theme={theme}>
                <HeaderBar />
            </ThemeProvider>
        );

        const { container } = render(<Page />);

        component = container;
    })

    describe('When it initializes', () => {
        it('matches snapshot', () => {
            expect(component).toMatchSnapshot();
        });
    });
});
