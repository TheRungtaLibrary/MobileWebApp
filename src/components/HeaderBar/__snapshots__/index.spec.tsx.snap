// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <HeaderBar /> When it initializes matches snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-i9gxme"
  >
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic sc-bdvvtL eoUkvj css-hip9hq-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-10ltm04-MuiToolbar-root"
      >
        <div
          class="MuiBox-root css-gj1lgi"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 sc-gsDKAQ keApQy css-ahj2mt-MuiTypography-root"
          >
            Car Photos
          </p>
        </div>
      </div>
    </header>
  </div>
</div>
`;
